syntax = "proto3";

package bloock;

option go_package = "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto";
option java_package = "com.bloock.sdk.bridge.proto";

import "shared.proto";
import "config.proto";


service KeyService {
    rpc GenerateLocalKey(GenerateLocalKeyRequest) returns (GenerateLocalKeyResponse);
    rpc GenerateManagedKey(GenerateManagedKeyRequest) returns (GenerateManagedKeyResponse);
}

message GenerateLocalKeyRequest {
    ConfigData config_data = 1;
    string key_type = 2;
    optional string key_curve = 3;
    optional string key_size = 4;
}

message GenerateLocalKeyResponse {
    string key = 1;
    optional string private_key = 2;
    optional Error error = 3;
}

message GenerateManagedKeyRequest {
    ConfigData config_data = 1;
    int32 protection = 2;
    string key_type = 3;
    optional string key_curve = 4;
    optional string key_size = 5;
    optional int64 expiration = 6;
}

message GenerateManagedKeyResponse {
    string key = 1;
    int32 protection = 2;
    string key_type = 3;
    optional string key_curve = 4;
    optional string key_size = 5;
    optional string name = 6;
    optional int64 expiration = 7;
    optional Error error = 8;
}
